admin-dev/themes/new-theme/js/pages/product/components/dropzone/Dropzone.vue

```diff
@@ -494,7 +494,7 @@
           const imageElement = document.querySelector(
             DropzoneMap.savedImage(newImage.image_id),
           );
-          imageElement.src = newImage.image_url;
+          imageElement.src = newImage.image_url + '?' + Math.random();
```

Comment:
What's the behavior behind this? It's because of browser's cache?

Reply:
yes, I think is the browser cache, you can force the browser to renew the new image url after the image uploaded.

Reply:
@PierreRambaud wdyt? I really appreciate this little trick

Reply:
If we want to do something properly we have to get the image checksum instead of a random number.
Using `Math.random` can return twice the same number.

Reply:
checksum is the best solution, if Math.random() may return a repeat number, we can also change **Math.random()** to **new Date().getTime()**, because all times a different, so this can avoid the same number repeat.

Reply:
If the image is only load once, why not, otherwise, if it has to be always displayed, we should use the checksum method

Reply:
> If the image is only load once, why not, otherwise, if it has to be always displayed, we should use the checksum method

Looks like it doesn't have to be always displayed, it's only on replace, in order to kick the browser's cache out which is faster, but maybe every API endpoints can just provide the URL with image checksum : ping @jolelievre

Reply:
> > If the image is only load once, why not, otherwise, if it has to be always displayed, we should use the checksum method
> 
> Looks like it doesn't have to be always displayed, it's only on replace, in order to kick the browser's cache out which is faster, but maybe every API endpoints can just provide the URL with image checksum : ping @jolelievre

agree with you, I come across that at product list page, the image is also not updated after image replacement because of the browser's cache, so maybe we have to makes some change at the image core API.